src/modules/blog/blog.service.ts

```diff
@@ -0,0 +1,116 @@
+import { NotFoundException } from './../../shared/exceptions/http.exception';
+import { Repository } from 'typeorm';
+import {
+  BadRequestException,
+  InternalServerErrorException,
+} from '../../shared/exceptions/http.exception';
+import { Blog } from './blog.entity';
+import { CreateBlogDto } from './dto/create-blog.dto';
+import { UpdateBlogDto } from './dto/update-blog.dto';
+import { v4 as uuidv4 } from 'uuid';
+
+
+export class BlogService {
+  constructor(private blogRepository: Repository<Blog>) {}
+
+  async createBlog(createBlogDto: CreateBlogDto) {
+    try {
+      const { authorId, title, content, tags } = createBlogDto;
+
+      const existingBlog = await this.blogRepository.findOne({ where: { title } });
+      if (existingBlog) {
+        throw new BadRequestException('Blog with this title already exists');
+      }
+
+      const newBlogId = uuidv4();
+
+      const newBlog = this.blogRepository.create({
+        id: newBlogId,
+        authorId,
+        title,
+        content,
+        tags,
+      });
+
+      await this.blogRepository.save(newBlog);
+
+      return newBlog;
+    } catch (error) {
+      if (error instanceof BadRequestException) {
+        throw error;
+      }
+      throw new InternalServerErrorException('Creating blog failed');
+    }
+  }
+
+  async getAllBlogs() {
+    try {
+      const blogs = await this.blogRepository.find();
+      if (!blogs.length) {
+        return {message: 'Blogs are empty!'}
+      }
+      return blogs;
+    } catch (error) {
+      throw new InternalServerErrorException('Something went wrong');
+    }
+  }
+
+  async getBlogById(id:string) {
+    try {
+      const blog = await this.blogRepository.findOne({ where: { id }});
+      if (!blog) {
+        return {message: 'Blog not found'}
```

Comment:
You have to throw error here. and try to make centralized error handler which classifies and returns user friendly error to user. I have seen you used trycatch in controller. In order to catch this error in controller you have to throw it here.